File: DemoProjectKMP-Public/Start/iOS/iOS_footballmatches/iOS_footballmatches/domain/LeagueRepositoryImpl.swift
//
//  LeagueRepository.swift
//  iOS_footballmatches
//
//  Created by Menes  SIMEU on 06/06/2024.
//

import Foundation

import Foundation

class LeagueRepositoryImpl : LeagueRepository {
    func getLeagues(completion: @escaping (Result<[League], Error>) -> Void) {
        NetworkService.shared.getLeagues { result in
            completion(result)
        }
    }
}

File: DemoProjectKMP-Public/End/KMP_footballmatches/shared/src/commonMain/kotlin/com/example/kmp_footballmatches/domain/LeagueRepositoryImpl.kt
package com.example.kmp_footballmatches.domain

import com.example.kmp_footballmatches.data.api.ApiKtorService
import com.example.kmp_footballmatches.data.model.League
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.IO
import kotlinx.coroutines.withContext
/**
 * Created by Menes SIMEU on 10/06/2024.
 * Volvo MoCE Android Team member
 */
class LeagueRepositoryImpl : LeagueRepository {
    private val apiService = ApiKtorService()


    override suspend fun getLeagues(): List<League> {
        return withContext(Dispatchers.IO) {
            apiService.getLeagues().data
        }
    }
}
